### PR TITLE
Week4

### PR DESCRIPTION
**Pull request by RimSebai**

### CODE REVIEW

File: homework-submission/blog/src/App.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      },
+      body: JSON.stringify(comment),
+    })
+      .then(comment => comment.json())
+      .then(comment => {
+        this.setState({ data: [...this.state.data, comment] });
+      });
+  };
+
+  handleUpdate = (updatedItem, id) => {
```
Comment (jalal75) [inline homework-submission/blog/src/App.js]: why do you need two parameters
you have the id inside the updatedItem :
( updatedItem._id )

File: homework-submission/blog/src/App.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    fetch(`https://hyf-react-api.herokuapp.com/blog/comments/${id}`, {
+      method: 'PATCH',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(updatedItem),
+    })
+      .then(response => response.json())
+      .then(response => {
+        this.state.data.map(comment => {
```
Comment (jalal75) [inline homework-submission/blog/src/App.js]: costly code...

File: homework-submission/blog/src/App.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+          if (comment._id === response._id) {
+            response.text = comment.text;
+            response.date = comment.date;
+            response.author.firstName = comment.author.firstName;
+            response.author.lastName = comment.author.lastName;
+          }
```
Comment (jalal75) [inline homework-submission/blog/src/App.js]: why this....???????????????

File: homework-submission/blog/src/App.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+          return comment;
+        });
+        this.reset();
```
Comment (jalal75) [inline homework-submission/blog/src/App.js]: you are fetching the all data, just for one updated item and that's not good
try to find another solution

File: homework-submission/blog/src/Comments.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import Update from './Update';
+
+class Comments extends React.Component {
+  state = {
+    update: false,
+    number: 0,
```
Comment (jalal75) [inline homework-submission/blog/src/Comments.js]: it is a good improve to show the number of like, but it is not saved in the Database in the server, means when refresh the page the counter for likes will be ZERO 
that is not good for the user experience

File: homework-submission/blog/src/Comments.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  };
+  handleEdit = () => {
+    this.setState({ update: true });
+  };
+  handleCancel = () => {
+    this.setState({ update: false });
+  };
+  handleUpdate = (updatedItem, id) => {
```
Comment (jalal75) [inline homework-submission/blog/src/Comments.js]: also id is available in updatedItem

File: homework-submission/blog/src/Comments.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    this.props.handleUpdate(updatedItem, id);
+    this.setState({ update: false });
+  };
+  handleClick = () => {
+    this.setState({ number: this.state.number + 1 });
+    this.props.item.isLiked = true;
+  };
```
Comment (jalal75) [inline homework-submission/blog/src/Comments.js]: and this function you are increasing the likes counter, but it should swap between like and unlike

File: homework-submission/blog/src/Comments.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+          value="delete"
+          onClick={() => this.props.handleDelete(this.props.item._id)}
+        />
+        <input type="button" value="edit" onClick={this.handleEdit} />
+        <input
+          type="button"
+          value={`like ${this.state.number}`}
+          onClick={() => this.handleClick()}
+        />
+      </div>
```
Comment (jalal75) [inline homework-submission/blog/src/Comments.js]: it is better  to make it with another component

**Comment by RimSebai** [comment]
Thanks Jalal for your feedback and answering all my questions through Slack! I will handle the errors as soon as possible.

**Pull request closed without merging**